### DISCUSSION TITLE
[@types/omggif] Should not reference `node`

### DISCUSSION THREAD

**Discussion by Lemmingh** [DefinitelyTyped/DefinitelyTyped#59738] [Issues with a @types package] [answered]
The beginning comment of `omggif` says:

<https://github.com/deanm/omggif/blob/0ec9069a525ec66223eadda3d62059d3685fd3b0/omggif.js#L23-L25>

> It does not rely on any specific underlying system, so should run in the browser, Node, or Plask.

And its code just uses an integer pointer (`p`) with the bracket notation to access `buf`.

Thus, the `buf` need not be a Node.js `Buffer`, and we can say it is expected to be:

An array-like object which represents an array of 8-bit unsigned integers.

---

From a practical perspective, the type of `buf` can be `Uint8Array | ArrayLike<number>`

```typescript
type GifBinary = Uint8Array | ArrayLike<number>;

export class GifWriter<TBuffer extends GifBinary = Uint8Array> {
    constructor(buf: TBuffer, /* ... */);

    getOutputBuffer(): TBuffer;
    setOutputBuffer(v: TBuffer): void;

    // ...
}

export class GifReader<TBuffer extends GifBinary = Uint8Array> {
    constructor(buf: TBuffer);

    // ...
}
```

Test cases:

```typescript
/// <reference types="node" />

const t = new Uint8Array(2 ** 20);
const b = Buffer.from(t);
const a = Array.from(t);

new GifWriter(t, 2, 2);
new GifWriter(b, 2, 2);
new GifWriter(a, 2, 2);
```

[a comment by an automated account is not shown]

**Comment by ffflorian** [comment, marked as answer]
Good idea @Lemmingh, thank you!

See https://github.com/DefinitelyTyped/DefinitelyTyped/pull/59739.
